packages/juce-blueprint/esm/lib/EventBridge.js

```diff
@@ -1,14 +1,36 @@
 import EventEmitter from 'events';
 
+class EventBridge extends EventEmitter {
+  constructor() {
+    super();
 
-const EventBridge = new EventEmitter();
+    this.emit = this.emit.bind(this);
 
-EventBridge.setMaxListeners(30);
+    this.emittingFromNative = false;
 
-// An internal hook for the native side, from which we propagate events through
-// the EventEmitter interface.
-__BlueprintNative__.dispatchEvent = function dispatchEvent(eventType, ...args) {
-  EventBridge.emit(eventType, ...args);
+    // TODO: Make constructor arg/configurable?
+    this.setMaxListeners(30);
+
+    __BlueprintNative__.dispatchEventFromNative = (eventType, ...args) => {
+      this.emittingFromNative = true;
+      this.emit(eventType, ...args);
+      this.emittingFromNative = false;
+    }
+  }
+
+  // Override EventEmitter.emit()
+  emit(eventType, ...args) {
+    if (this.emittingFromNative) {
+      //TODO: For some reason duktape does not like us using
+      //      super.emit() here. This may be something to do with 
+      //      how Babel transpiles such a call. Investigate. 
+      EventEmitter.prototype.emit.call(this, eventType, ...args);
```

Comment:
This was rather annoying and might be a useful thing to note for yourself @nick-thompson.

Reply:
Hm... indeed, that's annoying!